src/routes/profile.js

```diff
@@ -53,132 +55,113 @@ router.get('/mod/customavatar', isModMiddleware, (req, res) => {
-
-        if (
-          !foundReq.resLink.match(pattern) ||
-          !foundReq.resLink.match(pattern)
-        ) {
-          throw new Error(
-            `Invalid link provided: resLink: ${foundReq.resLink} spyLink: ${foundReq.spyLink}`,
-          );
-        }
-
-        await s3.rejectAvatarRequest(foundReq.resLink.match(pattern)[0]);
-        await s3.rejectAvatarRequest(foundReq.spyLink.match(pattern)[0]);
+router.post(
+  '/mod/ajax/processavatarrequest',
+  isModMiddleware,
+  async (req, res) => {
+    const avatarReq = await avatarRequest.findById(req.body.avatarreqid);
+    const userRequestingAvatar = await User.findOne({
+      usernameLower: avatarReq.forUsername.toLowerCase(),
+    });
 
-        foundReq.resLink = null;
-        foundReq.spyLink = null;
+    const modWhoProcessed = req.user;
 
-        User.findOne({ usernameLower: foundReq.forUsername.toLowerCase() })
-          .populate('notifications')
-          .exec((err, foundUser) => {
-            if (err) {
-              console.log(err);
-            } else {
-              let str = `Your avatar request was rejected by ${foundReq.modWhoProcessed}.`;
+    if (req.body.decision === true || req.body.decision === 'true') {
+      avatarReq.resLink = await s3.approveAvatarRefactorFilePath(
+        avatarReq.resLink,
+      );
+      avatarReq.spyLink = await s3.approveAvatarRefactorFilePath(
+        avatarReq.spyLink,
+      );
+      avatarReq.markModified('resLink');
+      avatarReq.markModified('spyLink');
 
-              if (foundReq.modComment) {
-                str += ` Their comment was: ${foundReq.modComment}`;
-              }
+      await avatarReq.save();
 
-              console.log(`string: ${str}`);
+      userRequestingAvatar.avatarImgRes = avatarReq.resLink;
+      userRequestingAvatar.avatarImgSpy = avatarReq.spyLink;
+      userRequestingAvatar.markModified('avatarImgRes');
+      userRequestingAvatar.markModified('avatarImgSpy');
 
-              createNotification(foundUser._id, str, '#', req.user.username);
-            }
-          });
-      } else {
-        console.log(
-          `error, decision isnt anything recognisable...: ${req.body.decision}`,
-        );
-        return;
-      }
+      await userRequestingAvatar.save();
 
-      const modUser = req.user;
-      // Create mod log - Doesn't need to be async
-      ModLog.create({
-        type: 'avatar',
-        modWhoMade: {
-          id: modUser._id,
-          username: modUser.username,
-          usernameLower: modUser.usernameLower,
-        },
-        data: {
-          modComment: req.body.modcomment,
-          approved: req.body.decision,
-          username: foundReq.forUsername,
-          msgToMod: foundReq.msgToMod,
-          resLink: foundReq.resLink,
-          spyLink: foundReq.spyLink,
-        },
-        dateCreated: new Date(),
-      });
+      // TODO-kev: Fundamental flaw in createNotification. Passing non unique link #
```

Comment:
Yep, as discussed let's remove the deduping logic in createNotification.js in a different PR.